tests/generation/test_utils.py

```diff
@@ -1372,6 +1373,26 @@ def test_assisted_decoding_sample(self):
 
             self._check_outputs(output_assisted, input_ids, model.config, use_cache=True)
 
+    def test_prompt_lookup_decoding_stops_at_eos(self):
```

Comment:
to make this test clearer and more robust, could we first get the shape of the output_prompt_lookup without an eos token id injected, and then show the difference once we add it?

Reply:
This indicates to me we're not testing or not able to test the right thing: the tests is called `test_prompt_lookup_decoding_stops_at_eos`, but if eos being present or not doesn't change anything I don't see how we're ensuring this?

Reply:
I think we are testing that the buggy behavior is not there anymore, so its output is 10 when we trigger setting when bug was found. That is actually all we can test :(

Reply:
OK :(